remote/remote_test.go

```diff
@@ -20,6 +26,15 @@ import (
 )
 
 var dockerRegistry, readonlyDockerRegistry *h.DockerRegistry
+var count int
+
+type RemoteTestCase struct {
+	name        string
+	repo        string
+	statusCode  int
+	failedCount int
+	expected    int
```

Comment:
```suggestion
	expectedTries    int
```

Reply:
IDK, this seems clearer.

Reply:
What about something like this:

`
if err != nil {
	// some code ..
	if transportErr, ok := err.(*transport.Error); ok && len(transportErr.Errors) > 0 && is40x(err) {
		return emptyImage(platform)
	}
	// more code ...
}

func isRetryable(err error) bool {
	if err == io.EOF {
		return true
	}
	return is40x(err)
}

func is40x(err error) bool {
	if transportErr, ok := err.(*transport.Error); ok {
		if transportErr.StatusCode == http.StatusNotFound ||
			transportErr.StatusCode == http.StatusUnauthorized {
			return true
		}
	}
	return false
}
`

Reply:
Applied